backend/lib/edgehog/astarte/astarte.ex

```diff
@@ -22,7 +22,14 @@ defmodule Edgehog.Astarte do
   use Ash.Domain
 
   resources do
-    resource Edgehog.Astarte.Cluster
-    resource Edgehog.Astarte.Realm
+    resource Edgehog.Astarte.Cluster do
+      define :create_cluster, action: :create
+    end
+
+    resource Edgehog.Astarte.Realm do
+      define :fetch_realm_by_name, action: :by_name, args: [:name]
```

Comment:
Given that we're touching this definition, you should also be able to delete the `:by_name` action on the resource and rely on the default `:read` instead.
In addition, I would set `not_found_error?: true`, also for uniformity with the code interfaces already defined in other domains.

```suggestion
      define :fetch_realm_by_name, action: :read, get_by: :name, not_found_error?: true
```